extensions/sources/etcd/src/main/java/io/smallrye/config/source/etcd/EtcdConfigSource.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.config.source.etcd;
+
+import com.coreos.jetcd.Client;
+import com.coreos.jetcd.ClientBuilder;
+import com.coreos.jetcd.data.ByteSequence;
+import com.coreos.jetcd.data.KeyValue;
+import com.coreos.jetcd.kv.GetResponse;
+import io.smallrye.config.source.EnabledConfigSource;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import lombok.extern.java.Log;
+import org.eclipse.microprofile.config.Config;
+
+/**
+ * Etcd config source
+ * @author <a href="mailto:phillip.kruger@redhat.com">Phillip Kruger</a>
+ * 
+ * TODO: Add watcher
+ *          ByteSequence bsKey = ByteSequence.fromString(EMPTY);
+ *          Watch.Watcher watch = this.client.getWatchClient().watch(bsKey);
+ *          watch.listen();
+ */
+@Log
+public class EtcdConfigSource extends EnabledConfigSource {
+    
+    private static final String NAME = "EtcdConfigSource";
+
+    private Client client = null;
+    private final Config config;
+    
+    public EtcdConfigSource(){
+        super.initOrdinal(320);
+        this.config = getConfig();
+    }
+    
+    @Override
+    public Map<String, String> getPropertiesIfEnabled() {
+        Map<String,String> m = new HashMap<>();
+        
+        ByteSequence bsKey = ByteSequence.fromString(EMPTY);
+
+        CompletableFuture<GetResponse> getFuture = getClient().getKVClient().get(bsKey);
+        try {
+            GetResponse response = getFuture.get();
+            List<KeyValue> kvs = response.getKvs();
+
+            for(KeyValue kv:kvs){
+                String key = kv.getKey().toStringUtf8();
+                String value = kv.getValue().toStringUtf8();
+                m.put(key, value);
+            }
+        } catch (InterruptedException | ExecutionException ex) {
```

Comment:
Thread interrupt status is not preserved.  If you catch `InterruptedException` without rethrowing the same or another `InterruptedException`, you *must* call `Thread.currentThread().interrupt()` before returning or throwing another exception.  Never wrap an `InterruptedException`.

Reply:
Change to Thread.currentThread().interrupt()